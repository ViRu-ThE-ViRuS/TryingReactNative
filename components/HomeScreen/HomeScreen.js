import React, { Component } from 'react'
import { StyleSheet, View, Text, FlatList, ActivityIndicator } from 'react-native'
import { ListItem } from 'react-native-elements'

export default class HomeScreen extends Component {
    static navigationOptions = {
        title: 'Home',
    }

    constructor(props) {
        super(props);

        this.state = {
            isLoadingData: true,
            data: []
        };
    }

    getData() {
        fetch('https://randomuser.me/api/?results=15')
            .then((response) => response.json())
            .then((response) => {
                this.setState({
                    isLoadingData: false,
                    data: response.results
                });
            });
    }

    componentDidMount() {
        this.getData();
    }

    _renderItem = ({ item }) => {
        return (<ListItem
            roundAvatar
            title={`${item.name.first} ${item.name.last}`}
            subtitle={item.email}
            avatar={{ uri: item.picture.thumbnail }}
            onPress={() => this.props.navigation.navigate('Profile', { userProfile: item })} />)
    }

    _keyExtractor = (item, index) => item.email;

    render() {
        if (this.state.isLoadingData) {
            return (
                <View style={styles.container}>
                    <ActivityIndicator style={styles.loading} size='large' />
                </View>
            )
        } else {
            return (
                <View style={styles.container}>
                    <FlatList
                        data={this.state.data}
                        renderItem={this._renderItem}
                        keyExtractor={this._keyExtractor} />
                </View>
            )
        }
    }
}

const styles = StyleSheet.create({
    container: {
        flex: 1
    },
    loading: {
        flex: 1
    }
});