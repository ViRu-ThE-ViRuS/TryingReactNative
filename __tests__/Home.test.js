import React from 'react';
import HomeScreen from '../components/HomeScreen/HomeScreen';
import renderer from 'react-test-renderer';

test('renders correctly', () => {
    const tree = renderer.create(<HomeScreen />).toJSON()
    expect(tree).toMatchSnapshot()
})