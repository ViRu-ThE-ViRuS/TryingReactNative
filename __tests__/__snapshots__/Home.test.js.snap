// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <ActivityIndicator
    animating={true}
    color="#999999"
    hidesWhenStopped={true}
    size="large"
    style={
      Object {
        "flex": 1,
      }
    }
  />
</View>
`;
